src/app/core/entity/entity-actions/entity-anonymize.service.ts

```diff
@@ -77,11 +78,21 @@ export class EntityAnonymizeService extends CascadingEntityAction {
       entity.getSchema().get(key).dataType,
     );
 
-    entity[key] = await dataType.anonymize(
-      entity[key],
-      entity.getSchema().get(key),
-      entity,
-    );
+    let anonymizedValue;
+    if (entity.getSchema().get(key).dataArray) {
```

Comment:
Can this produce null-pointer errors?

Reply:
this is called from an iteration of all schema fields, so we can safely assume these are defined.

Thanks for flagging this, though. We do have issue on this from time to time ...